src/util/tr.js

```diff
@@ -635,7 +624,27 @@ const _tr = (msg, {
       }
     } else {
       window.clevertap.popupCurrentWzrkId = targetingMsgJson.wzrk_id
-      renderFooterNotification(targetingMsgJson)
+      if (targetingMsgJson.display.deliveryTrigger) {
+        if (targetingMsgJson.display.deliveryTrigger.inactive) {
+          triggerByInactivity(targetingMsgJson)
+        }
+        if (targetingMsgJson.display.deliveryTrigger.scroll) {
+          triggerByScroll(targetingMsgJson)
+        }
+        if (targetingMsgJson.display.deliveryTrigger.isExitIntent) {
+          exitintentObj = targetingMsgJson
+          window.document.body.onmouseleave = showExitIntent
+        }
+        // delay
+        if ((targetingMsgJson[DISPLAY].delay != null && targetingMsgJson[DISPLAY].delay > 0) || (targetingMsgJson[DISPLAY].deliveryTrigger.deliveryDelayed != null && targetingMsgJson[DISPLAY].deliveryTrigger.deliveryDelayed > 0)) {
+          const delay = targetingMsgJson[DISPLAY].deliveryTrigger.deliveryDelayed ? targetingMsgJson[DISPLAY].deliveryTrigger.deliveryDelayed : targetingMsgJson[DISPLAY].delay
+          setTimeout(() => {
+            renderFooterNotification(targetingMsgJson)
+          }, delay * 1000)
+        }
+      } else {
+        renderFooterNotification(targetingMsgJson)
+      }
```

Comment:
Why is targetingMsgJson.display referred in two different ways in single if condition?
targetingMsgJson.display and targetingMsgJson[DISPLAY]

Reply:
Since targetingMsgJson.display is checked multiple times create a variable for it and use that variable

Reply:
Sure let me add a variable for targetingMsgJson.display

Reply:
Done